### ISSUE TITLE
Adding regex to global allowlist does not ignore secret, but adding to rule's allowlist does

### ISSUE THREAD

**Issue by emmahsax** [zricethezav/gitleaks#715]
**Describe the bug**
Adding regex to global allowlist doesn't ignore the secret, but adding to rule specific allowlist does.

If I add 
```
[allowlist]
regexes = ['''6Ld6_CcaAAAAAOXP4F6Ze2M5mbeqFRSEN9dlUecn''']
```

to my config file, then it doesn't properly ignore that secret. But if I add
```diff
[[rules]]
id = "generic-api-key"
description = "Generic API Key"
regex = '''(?i)((key|api|token|secret|password)[a-z0-9_ .\-,]{0,25})(=|>|:=|\|\|:|<=|=>|:).{0,5}['\"]([0-9a-zA-Z\-_=]{8,64})['\"]'''
entropy = 3.7
entropyGroup = 4
+ [rules.allowlist]
+    regexes = ['''6Ld6_CcaAAAAAOXP4F6Ze2M5mbeqFRSEN9dlUecn''']
```

to my config file's rule, then it does properly ignore the secret.

This is the secret the output is showing:
```
{
        "Description": "Generic API Key",
        "StartLine": 49,
        "EndLine": 49,
        "StartColumn": 70,
        "EndColumn": 115,
        "Context": "          data-expired-callback=\"recaptchaExpiredCallback\" data-sitekey=\"6Ld6_CcaAAAAAOXP4F6Ze2M5mbeqFRSEN9dlUecn\"",
        "Secret": "key=\"6Ld6_CcaAAAAAOXP4F6Ze2M5mbeqFRSEN9dlUecn\"",
        "File": "path/file_name.html",
        "Commit": "",
        "Entropy": 4.643943,
        "Author": "",
        "Email": "",
        "Date": "",
        "Message": "",
        "Tags": [],
        "RuleID": "generic-api-key"
}
```

**To Reproduce**

1. Create a commit with a new test secret
2. Add the secret to the global allowlist
3. Does it fail?
4. Now add it to the specific rule's allowlist
5. Does it pass again?

**Expected behavior**

The global allowlist of regexes to ignore is truly global and ignores the secrets put there.

**Screenshots**

**Basic Info (please complete the following information):**
 - OS: Running via docker on mac
 - Gitleaks Version: v8.0.0

**Additional context**

cc @zricethezav

**Comment by zricethezav** [comment]
@emmahsax great catch! This is indeed a bug

**Comment by zricethezav** [comment]
released in v8.0.4, thanks @emmahsax
